crates/evm/evm/src/metrics.rs

```diff
@@ -75,20 +75,19 @@ pub struct ExecutorMetrics {
 }
 
 impl ExecutorMetrics {
-    fn metered<F, R, B>(&self, block: &RecoveredBlock<B>, f: F) -> R
+    fn metered<F, R>(&self, f: F) -> R
     where
-        F: FnOnce() -> R,
-        B: reth_primitives_traits::Block,
+        F: FnOnce() -> (u64, R),
     {
         // Execute the block and record the elapsed time.
         let execute_start = Instant::now();
-        let output = f();
+        let (gas_used, output) = f();
         let execution_duration = execute_start.elapsed().as_secs_f64();
 
         // Update gas metrics.
-        self.gas_processed_total.increment(block.header().gas_used());
-        self.gas_per_second.set(block.header().gas_used() as f64 / execution_duration);
-        self.gas_used_histogram.record(block.header().gas_used() as f64);
+        self.gas_processed_total.increment(gas_used);
+        self.gas_per_second.set(gas_used as f64 / execution_duration);
+        self.gas_used_histogram.record(gas_used as f64);
```

Comment:
perhaps we get rid of this function entirely and inline this instead, but can do that later